cmd/tailscale/cli/up.go

```diff
@@ -138,6 +142,12 @@ func (a upArgsT) getAuthKey() (string, error) {
 
 var upArgs upArgsT
 
+// Fields output when `tailscale up --json` is used.
+type upJsonOutput struct {
+	AuthURL string `json:",omitempty"` // Authentication URL of the form https://login.tailscale.com/a/0123456789
+	QR      string `json:",omitempty"` // a DataURL (base64) PNG of a QR code AuthURL
+}
```

Comment:
I think we want a final message that says success rather than assuming that EOF means success.

Likewise: failures, like bogus auth codes or bogus tags.

Reply:
Added:

```
    {
      "AuthURL": "https://login.tailscale.com/a/0123456789",
      "QR": "data:image/png;base64,iV...QmCC"
    }
    
 ...time passes, until I click on the login link...
 
    {
      "BackendState": "Running"
    }
```

It can end with BackendState Running or NeedsAuthorization. If I use invalid tags it ends up as Running but the admin panel says the tags are invalid. I don't think that is being communicated back to the client as part of the authentication flow.